File: app/src/main/java/com/wyang/study/ui/fragment_second/OkHttpFragment.java
package com.wyang.study.ui.fragment_second;

import android.widget.TextView;

import com.wyang.study.R;
import com.wyang.study.ui.base.BaseFragment;

import java.io.IOException;

import okhttp3.Call;
import okhttp3.Callback;
import okhttp3.Interceptor;
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

public class OkHttpFragment extends BaseFragment {
    TextView tv_text;

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_example;
    }

    @Override
    protected void initView() {
        tv_text = mRootView.findViewById(R.id.tv_text);

        OkHttpClient client = new OkHttpClient.Builder()
                .addNetworkInterceptor(new Interceptor() {
                    @Override
                    public Response intercept(Chain chain) {
                        return null;
                    }
                })
                .build();

        Request request = new Request.Builder()
                .url("http://www.baidu.com")
                .build();

        Call call = client.newCall(request);

        call.enqueue(new Callback() {
            @Override
            public void onFailure(Call call, IOException e) {

            }

            @Override
            public void onResponse(Call call, Response response) {
                getActivity().runOnUiThread(() -> {
                    try {
                        tv_text.setText(response.body().string());
                    } catch (IOException e) {
                        e.printStackTrace();
                    }
                });
            }
        });
    }
}

File: app/src/main/java/com/wyang/study/ui/fragment_second/OkHttpFragment.kt
package com.wyang.study.ui.fragment_second

import com.wyang.study.databinding.FragmentExampleBinding
import com.wyang.study.ui.base.BaseFragment
import okhttp3.*
import java.io.IOException

class OkHttpFragment : BaseFragment<FragmentExampleBinding>() {
    override fun getViewBinding(): FragmentExampleBinding {
        return FragmentExampleBinding.inflate(layoutInflater)
    }

    override fun initialize() {
        val client: OkHttpClient = OkHttpClient.Builder()
            .build()

        val request: Request = Request.Builder()
            .url("http://www.baidu.com")
            .build()

        val call = client.newCall(request)

        call.enqueue(object : Callback {
            override fun onFailure(call: Call, e: IOException) {}
            override fun onResponse(call: Call, response: Response) {
                requireActivity().runOnUiThread {
                    try {
                        mBinding?.tvText?.text = response.body!!.string()
                    } catch (e: IOException) {
                        e.printStackTrace()
                    }
                }
            }
        })
    }
}